release/nightly_tests/dataset/ray_sgd_training.py

```diff
@@ -529,9 +531,13 @@ def train(self):
     preprocessing_end_time = timeit.default_timer()
     print("Preprocessing time (s): ", preprocessing_end_time - e2e_start_time)
 
-    num_columns = len(train_dataset.schema().names)
-    # remove label column and internal Arrow column.
-    num_features = num_columns - 2
+    # filter label column and internal Arrow column.
+    def is_feature_column(column_name):
+        return column_name != "label" and not column_name.startswith("__")
+
+    num_features = len(
+        list(filter(lambda name: is_feature_column(name), train_dataset.schema().names))
```

Comment:
nit: lambda is unnecessary 
```suggestion
        list(filter(is_feature_column, train_dataset.schema().names))
```